src/services/permissions-checker.js

```diff
@@ -16,58 +16,144 @@ class PermissionsChecker {
 
   static expirationInSeconds = EXPIRATION_IN_SECONDS;
 
-  // This permissionsPerRendering is the cache, shared by all instances of PermissionsChecker.
-  static permissionsPerRendering = {};
+  static isrolesACLActivated;
+
+  // This permissions object is the cache, shared by all instances of PermissionsChecker.
+  static permissions = { collections: {}, renderings: {} };
+
 
   static cleanCache() {
-    PermissionsChecker.permissionsPerRendering = {};
+    PermissionsChecker.permissions = { collections: {}, renderings: {} };
   }
 
-  static getPermissions(renderingId) {
-    return PermissionsChecker.permissionsPerRendering[renderingId];
+  // In the teamACL format, all the permissions are stored by renderingId into "renderings".
+  // For the rolesACL format, the collections permissions are stored directly into "collections",
+  // and only their scopes are stored by renderingId into "renderings".
+  static getCollectionsPermissions(renderingId) {
+    if (PermissionsChecker.isrolesACLActivated) {
+      return PermissionsChecker.permissions.collections.data;
+    }
+    return PermissionsChecker.permissions.renderings[renderingId]
+      ? PermissionsChecker.permissions.renderings[renderingId].data
+      : null;
   }
 
-  static _setPermissionsInRendering(renderingId, permissions) {
-    PermissionsChecker.permissionsPerRendering[renderingId] = permissions;
+  _getScopePermissions(collectionName) {
+    if (PermissionsChecker.permissions.renderings[this.renderingId]
+      && PermissionsChecker.permissions.renderings[this.renderingId].data
+      && PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName]) {
+      return PermissionsChecker.permissions.renderings[this.renderingId].data[collectionName].scope;
+    }
+    return null;
+  }
+
+  static transformPermissionsFromOldToNewFormat(permissions) {
+    Object.keys(permissions).forEach((modelName) => {
+      const { collection } = permissions[modelName];
+      permissions[modelName].collection = {
```

Comment:
define a new object instead of reusing the same object.
For 2 reasons:
- Your function is modifying a parameter which is really bad. We don't want that.
- Your permissions will actually contain the new AND the old format 😅

Reply:
Not sure why it will contain the old format 🤔 
But sure it would be better to user a new object here! 🙏